### ISSUE TITLE
Teuchos: YAML ParameterList getParametersFromYamlString has no implementation

### ISSUE THREAD

**Issue by eric-c-cyr** [trilinos/Trilinos#1085]
The method getParametersFromYamlString doesn't appear to have any implementation (a search in Teuchos yields only the declaration). Is there a plan for this, or is reading YAML files the only way to produce a parameter list from YAML?

**Comment by jmgate** [comment]
@trilinos/teuchos

**Comment by mhoemmen** [comment]
@csiefer2 might know something about this -- did that never get moved from MueLu?

**Comment by csiefer2** [comment]
Either of you guys know? 
@brian-kelley @DrBooom

**Comment by brian-kelley** [comment]
I just looked at my working copy of the Yaml stuff as of Aug. 8, when it was still in MueLu. I made "updateParametersFromYaml (CString/String/File)" and "getParametersFromYamlFile", but no "get...FromString". The implementation would just be "update...FromString", except with an empty list and overwrite = true. Would you like me to do that?

**Comment by jjellio** [comment]
I ran into this last night. I glanced at the YAML source and added the following to my own project's code as a quick fix. Unfortunately, it didn't work for what I wanted. (see after the code snipplet for that)


```
#include <Teuchos_YamlParser_decl.hpp>
namespace Teuchos {

TEUCHOSPARAMETERLIST_LIB_DLL_EXPORT
RCP<ParameterList> getParametersFromYamlString(const std::string &yamlStr)
{
  Teuchos::RCP<Teuchos::ParameterList> params = YAMLParameterList::parseYamlText(yamlStr);
  return (params);
}

}
```


What I was trying to do, was to take a Teuchos::describe( )'s output, and parse it into YAML. I don't recall the exact error I was getting, something about the structure of the YAML being invalid. This would be a separate issue though.

**Comment by brian-kelley** [comment]
Looks like @ibaned fixed this in c570303b (thanks!), so I'll close this.
